tests/tests_integration/test_api/test_data_modeling/test_containers.py

```diff
@@ -121,3 +123,26 @@ def test_dump_json_serialize_load(self, movie_containers: ContainerList) -> None
         container_json = json.dumps(container_dump)
         container_loaded = Container.load(container_json)
         assert container == container_loaded
+
+    def test_create_container_with_unit_property(
+        self, cognite_client: CogniteClient, integration_test_space: Space
+    ) -> None:
+        unit_container = ContainerApply(
+            space=integration_test_space.space,
+            external_id="test_container_with_unit",
+            properties={"pressure": ContainerProperty(type=Float64(unit=UnitReference(external_id="pressure:bar")))},
+            used_for="node",
+        )
+
+        try:
+            created = cognite_client.data_modeling.containers.apply(unit_container)
+
+            retrieved = cognite_client.data_modeling.containers.retrieve(created.as_id())
+
+            assert retrieved is not None
+            pressure_type = retrieved.properties["pressure"].type
+            assert isinstance(pressure_type, Float64)
+            assert pressure_type.unit is not None
+            assert pressure_type.unit.external_id == "pressure:bar"
+        finally:
+            cognite_client.data_modeling.containers.delete(unit_container.as_id())
```

Comment:
@erlendvollset made some good points about why we don't want to create+destroy containers in our tests. Maybe it would suffice to move the creation to a fixture with the "create if missing" pattern, then use it for testing?